codec/src/test/java/io/netty5/handler/codec/compression/ZstdEncoderTest.java

```diff
@@ -52,34 +51,32 @@ public EmbeddedChannel createChannel() {
 
     @ParameterizedTest
     @MethodSource("largeData")
-    public void testCompressionOfLargeBatchedFlow(final ByteBuf data) throws Exception {
+    public void testCompressionOfLargeBatchedFlow(final Buffer data) throws Exception {
         final int dataLength = data.readableBytes();
-        int written = 0;
+        try (Buffer expected = data.copy()) {
+            Buffer in = data.readSplit(65535);
+            assertTrue(channel.writeOutbound(in));
 
-        ByteBuf in = data.retainedSlice(written, 65535);
-        assertTrue(channel.writeOutbound(in));
+            Buffer in2 = data.readSplit(dataLength - 65535);
+            assertTrue(channel.writeOutbound(in2));
```

Comment:
`in2` would contain all the remaining readable bytes of `data`, so can't we just pass `data` here?
If so, then we could also inline `in` above, so neither local variable is necessary.